eth/executionclient/execution_client_test.go

```diff
@@ -2,30 +2,175 @@ package executionclient
+	n, err := node.New(&node.Config{})
+
+	if err != nil {
+		t.Fatalf("can't create new node: %v", err)
+	}
+	// Create Ethereum Service
+	config := &ethconfig.Config{Genesis: genesis}
+	ethservice, err := eth.New(n, config)
+	if err != nil {
+		t.Fatalf("can't create new ethereum service: %v", err)
+	}
+
+	// Add required APIs
+	filterSystem := filters.NewFilterSystem(ethservice.APIBackend, filters.Config{})
+	n.RegisterAPIs([]rpc.API{{
+		Namespace: "eth",
+		Service:   filters.NewFilterAPI(filterSystem, false),
+	}})
+
+	// Start eth1 node
+	if err := n.Start(); err != nil {
+		t.Fatalf("can't start test node: %v", err)
+	}
+
+	go func() {
+		defer close(processedStream)
+		select {
+		case <-done:
+			return
+		case blocks := <-blockStream:
+			// Batch block processing because events are fired only after all blocks in the batch processed
+			// for _, block := range blocks {
+			// 	var blocksToProcess []*types.Block
+			// 	blocksToProcess = append(blocksToProcess, block)
+			// 	if _, err := ethservice.BlockChain().InsertChain(blocksToProcess); err != nil {
+			// 		return
+			// 	}
+			// 	processedStream <- blocksToProcess
+			// 	time.Sleep(delay)
+			// }
+			if _, err := ethservice.BlockChain().InsertChain(blocks); err != nil {
+				return
+			}
+			processedStream <- blocks
+		}
+	}()
+	return n, processedStream
+}
+
+// Generate blocks with transactions
+func generateInitialTestChain(t *testing.T, done <-chan interface{}, blockStream chan []*types.Block, n int) {
+	generate := func(i int, g *core.BlockGen) {
+		g.OffsetTime(5)
+		g.SetExtra([]byte("test"))
+		if i == 0 {
+			return
+		}
+		// Add contract deployment to the firs block
+		if i == 1 {
+			tx := types.MustSignNewTx(testKey, types.LatestSigner(genesis.Config), &types.LegacyTx{
+				Nonce:    uint64(i - 1),
+				Value:    big.NewInt(0),
+				GasPrice: big.NewInt(params.InitialBaseFee),
+				Gas:      100000,
+				Data:     ethcommon.FromHex(callableBin),
+			})
+			g.AddTx(tx)
+		} else {
+			// Transactions to Callable contract
+			tx := types.MustSignNewTx(testKey, types.LatestSigner(genesis.Config), &types.LegacyTx{
+				To:       &contractAddr,
+				Nonce:    uint64(i - 1),
+				Value:    big.NewInt(0),
+				GasPrice: big.NewInt(params.InitialBaseFee),
+				Gas:      100000,
+				// Call to function Call() which emits event Called()
+				Data:     ethcommon.FromHex("0x34e22921"),
+			})
+			g.AddTx(tx)
+		}
+	}
+	_, blocks, _ := core.GenerateChainWithGenesis(genesis, ethash.NewFaker(), n, generate)
+	go func() {
+		select {
+		case <-done:
+		case blockStream <- blocks:
+		}
+	}()
+}
+
+/*
+Example contract to test event emission:
+
+	pragma solidity >=0.7.0 <0.9.0;
+	contract Callable {
+		event Called();
+		function Call() public { emit Called(); }
+	}
+*/
+const callableAbi = "[{\"anonymous\":false,\"inputs\":[],\"name\":\"Called\",\"type\":\"event\"},{\"inputs\":[],\"name\":\"Call\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"}]"
```

Comment:
Will we need it? Can we delete it?

Reply:
Yes, lets remove it. I saved in case we would need to use ABI